client/core/bond.go

```diff
@@ -926,6 +928,135 @@ func (c *Core) monitorBondConfs(dc *dexConnection, bond *asset.Bond, reqConfs ui
 	})
 }
 
+// RedeemPrepaidBond redeems a pre-paid bond for a dcrdex host server.
+func (c *Core) RedeemPrepaidBond(appPW []byte, code []byte, host string, certI any) (tier uint64, err error) {
+	// Make sure the app has been initialized.
+	if !c.IsInitialized() {
+		return 0, fmt.Errorf("app not initialized")
+	}
+
+	// Check the app password.
+	crypter, err := c.encryptionKey(appPW)
+	if err != nil {
+		return 0, codedError(passwordErr, err)
+	}
+	defer crypter.Close()
+
+	var success, acctExists bool
+
+	c.connMtx.RLock()
+	dc, found := c.conns[host]
+	c.connMtx.RUnlock()
+	if found {
+		acctExists = !dc.acct.isViewOnly()
+		if acctExists {
+			if dc.acct.locked() { // require authDEX first to reconcile any existing bond statuses
+				return 0, newError(acctKeyErr, "acct locked %s (login first)", host)
+			}
+		}
```

Comment:
Can remove the nested if: `if acctExists && dc.acct.locked() { return 0, newError(acctKeyErr, "acct locked %s (login first)", host) }`